src/systems/log/LogPlayback.cc

```diff
@@ -133,53 +138,15 @@ LogPlayback::~LogPlayback()
 }
 
 //////////////////////////////////////////////////
-void LogPlaybackPrivate::Parse(EntityComponentManager &_ecm,
-    const msgs::Pose_V &_msg)
+void LogPlaybackPrivate::Parse(const msgs::Pose_V &_msg)
 {
-  // Maps entity to pose recorded
-  // Key: entity. Value: pose
-  std::map <Entity, msgs::Pose> idToPose;
-
-  for (int i = 0; i < _msg.pose_size(); ++i)
+  // save the new entity pose updates
+  this->recentEntityPoseUpdates.clear();
```

Comment:
Is it correct to clear the `recentEntityPoseUpdates`? `Parse(const msgs::Pose_V &_msg)` could be called multiple times from a single `Update` iteration, so would clearing the `map` miss some pose updates?

Reply:
That's a good point. Looking at the `LogPlayback::Update` function, it looks like there are two places where `Parse(const msgs::Pose_V &_msg)` could be called: lines 486 & 563. I haven't seen an issue of missing pose updates in the tests I ran, but maybe that's because the rendering updates at 60Hz, which means the naked eye could miss something that happened for a single frame (and actually, now that you bring this up, maybe this is why the `LogSystemTest.RecordAndPlayback` test that I mentioned in my comment [above](https://github.com/ignitionrobotics/ign-gazebo/pull/351#issuecomment-693677769) is failing).

How about I define a boolean flag in the `LogPlayback::Update` method that decides whether `clear()` should be called or not? `clear()` should only be called in the first `Parse(const msgs::Pose_V &_msg)` call for each `Update`, so the flag will initially be set to true, but once `Parse(const msgs::Pose_V &_msg)` has been called once, then the flag will be set to false for the rest of the `Update` iteration.

Reply:
Handling multiple `Parse` calls in a single `Update` has been addressed in commit [06ba47b](https://github.com/ignitionrobotics/ign-gazebo/pull/351/commits/06ba47be6a19ecee0e03cb93a8daf80250d22e9a).

Reply:
does it work if you just call  `this->dataPtr->recentEntityPoseUpdates.clear()` once before going into the while loop then you don't need to handle clearing the cache in `Parse` with `_clear`?

Reply:
Done in b07ebddf7d552e56030213228c0ad3c9384bab58

Reply:
> does it work if you just call `this->dataPtr->recentEntityPoseUpdates.clear()` once before going into the while loop then you don't need to handle clearing the cache in `Parse` with `_clear`?

I don't believe that this will work - see my comment [below](https://github.com/ignitionrobotics/ign-gazebo/pull/351#issuecomment-694315698).

Reply:
ok got it thanks for the explanation. That makes sense. I think it helps to document this behavior. Could you add a brief inline comment on this, maybe at where  `clearCachedPoseUpdates` is declared?

Reply:
Yeah, I left a comment in [dee2afe](https://github.com/ignitionrobotics/ign-gazebo/pull/351/commits/dee2afe3fd488201dd7a7aeb2bbd94113246d2f4). There is also another mention of the multiple parse calls in the [`Parse(const msgs::Pose_V &_msg, bool &_clear)` method documentation](https://github.com/ignitionrobotics/ign-gazebo/blob/dee2afe3fd488201dd7a7aeb2bbd94113246d2f4/src/systems/log/LogPlayback.cc#L73-L79). Do those comments seem to be helpful?

Reply:
thanks, yeah the comments are helpful.

Reply:
Great, I'll mark this as resolved (feel free to "unresolve" this conversation if any other issues arise).